linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -608,8 +612,13 @@ private final class ClassDefChecker(classDef: ClassDef,
 
       case _: LoadModule =>
 
-      case StoreModule(_, value) =>
-        checkTree(value, env)
+      case StoreModule() =>
+        if (!classDef.kind.hasModuleAccessor)
+          reportError(i"Illegal StoreModule inside class of kind ${classDef.kind}")
+        if (!env.inConstructor)
+          reportError(i"Illegal StoreModule outside of constructor")
+        if (env.thisType == NoType)
```

Comment:
I was thrown off by this. Maybe add a comment that this (no pun intended) can happen in a JS ctor before the super call?